datafusion/functions-aggregate/src/count.rs

```diff
@@ -179,6 +179,107 @@ impl Count {
         }
     }
 }
+fn get_primitive_type_accumulator(data_type: &DataType) -> Box<dyn Accumulator> {
```

Comment:
Minor: technically this does much more than primitive types

For example `Utf8` is not a primitive types

Maybe it could be more precisely be called `get_count_accumulator`